### PR TITLE
feat(console): friendly quota exhausted screen

### PR DESCRIPTION
**Pull request by eliobischof**
Implements the third AC *Console shows a page with ZITADEL-wide configurable text template with as much available template variable as possible if cookie quota depleted is true*

- Shows a dialog after a gRPC or an http call response with a resource exhausted code.
- The dialog shows a Continue button. If the environment.json has an instance_management_url property, that page is loaded on click, else the the current page is just reloaded. 
- If the environment.json hast the exhausted property set to true the dialog is shown. 

```[tasklist]

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] Critical parts are tested automatically
- [x] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [x] Functionality of the acceptance criteria is checked manually on the dev system.
```

### CODE REVIEW

File: console/src/app/services/interceptors/exhausted.http.interceptor.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export class ExhaustedHttpInterceptor implements HttpInterceptor {
 
   intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
     // Don't check the cookie for the environment.json
-    let cookie$ = req.url.endsWith('/assets/environment.json') ? of(undefined) : this.exhaustedSvc.checkCookie();
+    let cookie$ = req.url.endsWith('./assets/environment.json') ? of(undefined) : this.exhaustedSvc.checkCookie();
```
Comment (peintnermax) [inline console/src/app/services/interceptors/exhausted.http.interceptor.ts:16]: So is the idea that the cookie references to the environment.json? and that again to the instance url?

Comment (eliobischof) [reply console/src/app/services/interceptors/exhausted.http.interceptor.ts]: fixed it

**Comment by eliobischof** [comment]
@peintnermax it's ready for review again.
Thx

**Review by peintnermax — CHANGES_REQUESTED** [review]

File: console/src/app/services/environment.service.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  private get cookiesAreEnabled() {
+    try {
+      document.cookie = 'testcookie';
+      const canSetCookie = document.cookie.indexOf('testcookie') != -1;
+      document.cookie = 'testcookie; expires=Thu, 01 Jan 1970 00:00:00 UTC';
+      return canSetCookie;
+    } catch (e) {
+      return false;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (eliobischof) [reply console/src/app/services/environment.service.ts]: done

File: console/src/app/app.module.ts [hunk 3]
```diff
@@ -139,6 +141,8 @@ const authConfig: AuthConfig = {
     RoleGuard,
     UserGuard,
     ThemeService,
+    ExhaustedService,
+    ExhaustedHttpInterceptor,
```
[the remark this answers is not in the document]
Comment (eliobischof) [reply console/src/app/app.module.ts:147]: done

**Review by peintnermax — APPROVED** [review]

**Pull request merged**